src/databricks/labs/ucx/hive_metastore/table_migrate.py

```diff
@@ -15,42 +15,53 @@
 from databricks.labs.ucx.framework.crawlers import SqlBackend
 from databricks.labs.ucx.hive_metastore import TablesCrawler
 from databricks.labs.ucx.hive_metastore.mapping import Rule, TableMapping
-from databricks.labs.ucx.hive_metastore.tables import MigrationCount, Table
+from databricks.labs.ucx.hive_metastore.tables import MigrationCount, Table, TableMigrationType
 
 logger = logging.getLogger(__name__)
 
 
 class TablesMigrate:
+
     def __init__(
         self,
         tc: TablesCrawler,
         ws: WorkspaceClient,
         backend: SqlBackend,
         tm: TableMapping,
+        *,
+        table_migration_type: TableMigrationType = None,
```

Comment:
```suggestion
```
no way this can be a property - it requires changing all callsites.

Reply:
Fixed